File: src/main/java/kr/co/lunatalk/domain/payment/repository/PaymentRepositoryCustom.java
package kr.co.lunatalk.domain.payment.repository;

import kr.co.lunatalk.domain.payment.domain.PaymentStatus;

import java.time.LocalDateTime;

public interface PaymentRepositoryCustom {
	Long sumAmountByStatusAndApprovedAtBetween(
		PaymentStatus status,
		LocalDateTime start,
		LocalDateTime end
	);

	Long countByStatusAndApprovedAtBetween(
		PaymentStatus status,
		LocalDateTime start,
		LocalDateTime end
	);
}

File: lunatalk-core/src/main/kotlin/kr/co/lunatalk/domain/payment/repository/PaymentRepositoryCustom.kt
package kr.co.lunatalk.domain.payment.repository

import kr.co.lunatalk.domain.payment.domain.PaymentStatus
import java.time.LocalDateTime

interface PaymentRepositoryCustom {
    fun sumAmountByStatusAndApprovedAtBetween(
        status: PaymentStatus,
        start: LocalDateTime,
        end: LocalDateTime,
    ): Long

    fun countByStatusAndApprovedAtBetween(
        status: PaymentStatus,
        start: LocalDateTime,
        end: LocalDateTime,
    ): Long
}
